CONTRIBUTING.md

```diff
@@ -0,0 +1,237 @@
+
+:wrench: When progressing through these steps, refer to the following for troubleshooting:
+
+```bash
+cat ~/.pgrx/15.log
+```
+
+#### 3.1. Apply configurations
+
+Prior to compiling and running `pg_vector`, it's essential to update the `postgresql.conf` file.
+`pgrx` uses a Postgres version-specific data directory, each containing its own `postgresql.conf` file.
+The following example, utilizes Postgres version 15.
+If you're using a different version, please alter the file path value `data-<postgres-version>` and run the following:
+
+```bash
+<your-editor> ~/.pgrx/data-15/postgresql.conf
+```
+
+Within this document, add the following:
+
+```text
+shared_preload_libraries = 'pg_cron, vectorize'
+cron.database_name = 'postgres'
+vectorize.embedding_service_url = 'http://vector-serve:3000/v1/embeddings'
+```
+
+:wrench: Note: If your machine is running a MacOS, you may need to apply the following configurations to Cargo's config file:
+
+```
+<your-editor> ~/.cargo/config
+```
+
+```text
+[target.'cfg(target_os="macos")']
+# Postgres symbols won't be available until runtime
+rustflags = ["-Clink-arg=-Wl,-undefined,dynamic_lookup"]
+```
+
+#### 3.2. Clone and enter directory
+
+```bash
+git clone https://github.com/tembo-io/pg_vectorize.git
+
+cd pg_vectorize
+```
+
+#### 3.3. Install dependencies
+
+From within the pg_vectorize directory, run the following, which will install `pg_cron`, `pgmq`, and `pg_vector`:
+
+```bash
+make setup
+```
+
+#### 3.4. Compile and run `pg_vector`
+
+```bash
+make run
+```
+
+### 4. Confirm successful build
+
+#### 4.1. Check extension presence
+
+Once the above command is run, you will be brought into Postgres via `psql`.
+
+To list out the enabled extensions, run:
+
+```sql
+\dx
+```
+```text
+                                      List of installed extensions
+    Name    | Version |   Schema   |                             Description
+------------+---------+------------+---------------------------------------------------------------------
+ pg_cron    | 1.6     | pg_catalog | Job scheduler for PostgreSQL
+ pg_partman | 4.7.3   | public     | Extension to manage partitioned tables by time or ID
+ pgmq       | 1.1.1   | pgmq       | A lightweight message queue. Like AWS SQS and RSMQ but on Postgres.
+ plpgsql    | 1.0     | pg_catalog | PL/pgSQL procedural language
+ vector     | 0.6.0   | public     | vector data type and ivfflat and hnsw access methods
+ vectorize  | 0.10.1  | vectorize  | The simplest way to do vector search on Postgres
+(6 rows)
+```
+
+#### 4.2 Change embedding service url to localhost
+
+```sql
+SHOW vectorize.embedding_service_url;
+```
+```text
+   vectorize.embedding_service_url
+-------------------------------------
+ http://vector-serve:3000/v1/embeddings
+(1 row)
+```
+
+We have to use local host
+```
+ALTER SYSTEM SET vectorize.embedding_service_url TO 'http://localhost:3000/v1/embeddings';
+```
+
+Upon making this change, run:
+
+```sql
+SELECT pg_reload_conf();
+```
```

Comment:
We dont need this step if we set the url properly in postgresql.conf in the first place.